pkg/registry/batch/job/strategy.go

```diff
@@ -184,9 +184,10 @@ func validationOptionsForJob(newJob, oldJob *batch.Job) batchvalidation.JobValid
 		notStarted := oldJob.Status.StartTime == nil
 		opts.AllowMutableSchedulingDirectives = utilfeature.DefaultFeatureGate.Enabled(features.JobMutableNodeSchedulingDirectives) &&
 			suspended && notStarted
-
 	}
 
+	// Elastic indexed jobs (mutable completions iff updated parallelism == updated completions)
+	opts.AllowElasticIndexedJobs = utilfeature.DefaultFeatureGate.Enabled(features.ElasticIndexedJob)
```

Comment:
nit: move this up into the `batchvalidation.JobValidationOptions{` declaration, to keep all the things that don't depend on `oldJob` together